modules/ml/src/main/java/org/apache/ignite/ml/environment/parallelism/ParallelismStrategy.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.ml.environment.parallelism;
+
+import org.apache.ignite.ml.math.functions.IgniteSupplier;
+
+/**
+ * Specifies the behaviour of processes in ML-algorithms that can may be parallelized such as parallel learning in
+ * bagging, learning submodels for One-vs-All model, Cross-Validation etc.
+ */
+public interface ParallelismStrategy {
+    public enum Type {
```

Comment:
Maybe we should add here "CUSTOM_POOL" with validation of custom pool checks?

Reply:
It seems to be interesting but CUSTOM_POOL require more information about instantiation it and that's why creation cannot be performed only by enum-value. If user wants custom pool then he/she may implements ParallelismStrategy with custom pool and use .withParallelismStrategy(strategy) method.